src/Toolkit/Toolkit.Maui/BasemapGallery/BasemapGallery.cs

```diff
@@ -185,7 +205,11 @@ private static void AvailableBasemapsChanged(BindableObject sender, object oldVa
     public IList<BasemapGalleryItem>? AvailableBasemaps
     {
         get => GetValue(AvailableBasemapsProperty) as IList<BasemapGalleryItem>;
-        set => SetValue(AvailableBasemapsProperty, value);
+        set
+        {
+            SetValue(AvailableBasemapsProperty, value);
+            _loadCancellationTokenSource?.Cancel();

```

Comment:
I think it would be cleaner to move all the cancellation logic into the `BasemapControllerGallery`, so it's is shared by all users of it.